projects/etos_suite_runner/src/etos_suite_runner/lib/suite.py

```diff
@@ -318,6 +319,67 @@ def sub_suite_environments(self) -> Iterator[dict]:
             self._record_exception(exc)
             raise exc
 
+    def _sub_suite_environments_from_kubernetes(self) -> Iterator[dict]:
+        """All sub suite environments from Kubernetes.
+
+        Each sub suite environment is an environment for the sub suites to execute in.
+        """
+        environments = []
+        timeout = time.time() + self.etos.config.get("WAIT_FOR_ENVIRONMENT_TIMEOUT")
+        while time.time() < timeout:
+            time.sleep(5)
+            for environment in self.params.environments:
+                if environment.spec.sub_suite_id in environments:
+                    continue
+
+                # Send eiffel event for the ETR.
+                event = EiffelEnvironmentDefinedEvent()
+                event.meta.event_id = environment.metadata.name
+                url = f"{os.getenv('ETOS_API')}/v1alpha/testrun/{environment.metadata.name}"
+                self.etos.events.send(
+                    event,
+                    {"CONTEXT": self.test_suite_started_id},
+                    {"name": environment.spec.name, "uri": url},
+                )
+
+                environments.append(environment.spec.sub_suite_id)
+                executor = environment.spec.executor
+                yield {
+                    "executor": executor,
+                    "id": environment.spec.sub_suite_id,
+                    "name": environment.spec.name,
+                }
+            status = self.params.get_status()
+            if status.get("status") == "FAILURE":
+                exc = EnvironmentProviderException(
+                    status.get("error"), self.etos.config.get("task_id")
+                )
+                self._record_exception(exc)
+                raise exc
+            if status.get("status") == "SUCCESS" and len(environments) > 0:
+                return
+        else:  # pylint:disable=useless-else-on-loop
+            exc = TimeoutError(
+                f"Timed out after {self.etos.config.get('WAIT_FOR_ENVIRONMENT_TIMEOUT')} seconds."
+            )
+            self._record_exception(exc)
+            raise exc
+
+    @property
+    def sub_suite_environments(self) -> Iterator[dict]:
+        """All sub suite environments from the environment provider.
+
+        Each sub suite environment is an environment for the sub suites to execute in.
+        """
+        self.logger.debug(
+            "Start collecting sub suite definitions (timeout=%ds).",
+            self.etos.config.get("WAIT_FOR_ENVIRONMENT_TIMEOUT"),
+        )
+        if os.getenv("IDENTIFIER") is not None:
```

Comment:
IDENTIFIER is set in v1+? If so, please add a comment.

Reply:
Moved to esr_parameters.